docs/developer-guide/plugins.md

```diff
@@ -16,6 +16,8 @@ var myPluginRule = stylelint.createPlugin(myPluginRuleName, function(expectation
 
 `stylelint.createPlugin(ruleName, ruleFunction)` ensures that your plugin will be setup properly alongside other rules. *Make sure you document your plugin's rule name for users, because they will need to use it in their config.*
 
+Your plugin's rule name must be prefixed e.g. `your-prefix/your-rule-name`. If your plugin exports a single rule then prefix your rule name with just a slash (e.g. `/your-rule-name`)
```

Comment:
Let's call it namespacing instead of prefixing. I think that makes the reasoning more apparent.

> Your plugin's rule name must be namespaced, e.g. `your-namespace/your-rule-name`. If your plugin exports a single rule, you may simply namespace your rule a slash (e.g. `/your-rule-name`). This namespace ensures that plugin rules will never clash with core rules.